client/search-ui/src/results/sidebar/SearchAggregations.tsx

```diff
@@ -0,0 +1,221 @@
+import { FC, useCallback, useMemo } from 'react'
+
+import { mdiArrowExpand, mdiPlus } from '@mdi/js'
+import { ParentSize } from '@visx/responsive'
+import { getTicks } from '@visx/scale'
+import { AnyD3Scale } from '@visx/scale/lib/types/Scale'
+import { useHistory, useLocation } from 'react-router'
+
+import { ButtonGroup, Button, Icon, BarChart } from '@sourcegraph/wildcard'
+
+import { LANGUAGE_USAGE_DATA, LanguageUsageDatum } from './search-aggregation-mock-data'
+
+import styles from './SearchAggregations.module.scss'
+
+interface URLStateOptions<State> {
+    urlKey: string
+    deserializer: (value: string | null) => State
+    serializer: (state: State) => string
+}
+
+type SetStateResult<State> = [state: State, dispatch: (state: State) => void]
+
+/**
+ * React hook analog standard react useState hook but with synced value with URL
+ * through URL query parameter.
+ */
+function useSyncedWithURLState<State>(options: URLStateOptions<State>): SetStateResult<State> {
+    const { urlKey, serializer, deserializer } = options
+    const history = useHistory()
+    const { search } = useLocation()
+
+    const urlSearchParameters = useMemo(() => new URLSearchParams(search), [search])
+    const queryParameter = useMemo(() => deserializer(urlSearchParameters.get(urlKey)), [
+        urlSearchParameters,
+        urlKey,
+        deserializer,
+    ])
+
+    const setNextState = useCallback(
+        (nextState: State) => {
+            urlSearchParameters.set(urlKey, serializer(nextState))
+
+            history.replace({ search: `?${urlSearchParameters.toString()}` })
+        },
+        [history, serializer, urlKey, urlSearchParameters]
+    )
+
+    return [queryParameter, setNextState]
+}
+
+enum AggregationModes {
+    Repository = 'repo',
+    FilePath = 'file',
+    Author = 'author',
+    CaptureGroups = 'groups',
+}
+
+const AGGREGATION_MODE_URL_KEY = 'groupBy'
+
+const aggregationModeDeserializer = (serializedValue: string | null): AggregationModes => {
+    switch (serializedValue) {
+        case 'repo':
+            return AggregationModes.Repository
+        case 'file':
+            return AggregationModes.FilePath
+        case 'author':
+            return AggregationModes.Author
+        case 'groups':
```

Comment:
We decided we'll only use the first capture group, so this should be singular. I'd also recommend spelling `captureGroup` out, because `groupBy=group` could be confusing.
```suggestion
        case 'captureGroup':
```

Reply:
Agreed, I changed serialise/deserialise functions. Thanks.